Source/ACE/Entity/Landblock.cs

```diff
@@ -186,6 +188,39 @@ public void RemoveWorldObject(ObjectGuid objectId, bool adjacencyMove)
             }
         }
 
+        /// <summary>
+        /// Handle the QueryHealth action between the source Object and its target
+        /// </summary>
+        public void HandleQueryHealth(Session source, ObjectGuid targetId)
+        {
+            if (targetId.IsPlayer())
+            {
+                Player pl = null;
+
+                lock (objectCacheLocker)
+                {
+                    if (this.worldObjects.ContainsKey(targetId))
+                        pl = (Player)this.worldObjects[targetId];
+                }
+                if (pl == null)
+                {
+                    // check adjacent landblocks for the targetId
+                    foreach (var block in adjacencies)
+                    {
+                        lock (objectCacheLocker)
+                        {
+                            if (block.Value.worldObjects.ContainsKey(targetId))
+                                pl = (Player)this.worldObjects[targetId];
+                        }
+                    }
+                }
+                float healthPercentage = (float)pl.Health.Current / (float)pl.Health.MaxValue;

```

Comment:
pl can be null at this point, too (player portals off the block).  Just a quick null check will prevent exceptions.